areas/appservice/tests/AzureMcp.AppService.UnitTests/Commands/Database/DatabaseAddCommandTests.cs

```diff
@@ -0,0 +1,362 @@
+        var response = await command.ExecuteAsync(context, args);
+
+        // Assert
+        Assert.NotNull(response);
+        Assert.Equal(200, response.Status);
+
+        await _appServiceService.Received(1).AddDatabaseAsync(
+            appName,
+            resourceGroup,
+            databaseType,
+            databaseServer,
+            databaseName,
+            Arg.Any<string>(),
+            subscription,
+            tenant,
+            Arg.Any<RetryPolicyOptions>());
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_MissingRequiredParameter_ReturnsErrorResponse()
+    {
+        // Arrange
+        var command = new DatabaseAddCommand(_logger);
+        var args = command.GetCommand().Parse([
+            "--subscription", "sub123",
+            "--resource-group", "rg1"
+            // Missing required parameters
+        ]);
+        var context = new CommandContext(_serviceProvider);
+
+        // Act
+        var response = await command.ExecuteAsync(context, args);
+
+        // Assert
+        Assert.NotNull(response);
+        Assert.NotEqual(200, response.Status);
+
+        await _appServiceService.DidNotReceive().AddDatabaseAsync(
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<RetryPolicyOptions>());
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_ServiceThrowsException_ReturnsErrorResponse()
+    {
+        // Arrange
+        var subscription = "sub123";
+        var resourceGroup = "rg1";
+        var appName = "test-app";
+        var databaseType = "SqlServer";
+        var databaseServer = "test-server.database.windows.net";
+        var databaseName = "test-db";
+
+        _appServiceService
+            .When(x => x.AddDatabaseAsync(
+                Arg.Any<string>(),
+                Arg.Any<string>(),
+                Arg.Any<string>(),
+                Arg.Any<string>(),
+                Arg.Any<string>(),
+                Arg.Any<string>(),
+                Arg.Any<string>(),
+                Arg.Any<string>(),
+                Arg.Any<RetryPolicyOptions>()))
+            .Do(x => throw new InvalidOperationException("Service error"));
+
+        var command = new DatabaseAddCommand(_logger);
+        var args = command.GetCommand().Parse([
+            "--subscription", subscription,
+            "--resource-group", resourceGroup,
+            "--app-name", appName,
+            "--database-type", databaseType,
+            "--database-server", databaseServer,
+            "--database-name", databaseName
+        ]);
+        var context = new CommandContext(_serviceProvider);
+
+        // Act
+        var response = await command.ExecuteAsync(context, args);
+
+        // Assert
+        Assert.NotNull(response);
+        Assert.NotEqual(200, response.Status);
```

Comment:
I think the response code on an error is `500`. We should probably compare against that. Not `200` could be `201`, which is still successful according to the REST spec.